js/build.js

```diff
@@ -133,6 +134,25 @@
         })
       }
 
+      /**
+       * A function that searches for a personal style of the widget
+       *
+       * @returns {object/null} returns a color object or false if there is no personal color used for this widget
+       */
+      function getPersonalColors() {
+        var widgetData = window.__widgetData;
```

Comment:
@hcarneiro @squallstar Can either or both of you review the logic of this? I think this is trying to retrieve the custom styles for a specific widget. I'm also surprised that we don't have a JS API for this.

```js
function (id) {
  // id @number Widget ID
  // return @object Theme settings for the given widget ID
}
```

Reply:
@tonytlwu I actually recommended this code to Upplabs because we don't have a JS API to get the theme instance settings. It's quite an uncommon use case, hence why no specific JS API is available.

Reply:
@squallstar Can you review and verify then?

Separately, I think we need a JS API just like we created `Fliplet.Theme.Current.get()` in the first place. It was created for this component. And other components that need theme settings in JS would also need it. If it's more complex than to adapt this function, then leave it. It's a lost opportunity I guess.

Reply:
@tonytlwu I am adding a new method as documented here:

```
Fliplet.Themes.Current.getInstance()
```

![img](https://user-images.githubusercontent.com/574210/80227506-fe94f480-864d-11ea-9333-b1a14498eb42.png)

@YaroslavOvdii this will be on staging later today. Once available, please use it to refactor part of your code to make use of it.

Reply:
@squallstar refactored code according to the new API method.
May I add small suggestion?
Can we add to the API function that returns only custom styles for certain UUID?
Because now we receiving entire theme objects and looking for custom styles manually.
For example something like this:
```
      getCustomStyles: function (uuid) {
        var instances = Fliplet.Themes.Current.getInstance().data.widgetInstances;
        return _.find(instances, function(instance) {
            return _.get(instance, 'uuid') === uuid;
        });
      }
```

Reply:
@YaroslavOvdii are you suggesting this because you will need this code more than once on different repos?

Reply:
@squallstar Yes, this code will be needed in all chart repos.
And in the future in all repos where we need to update custom styles from the JS.

Reply:
@YaroslavOvdii I see, OK I am going to add the following JS API:

```js
var instance = Fliplet.Themes.Current.getSettingsForWidgetInstance(uuid);
// instance { id, component, uuid, values }
```

This will be live on staging in around 1h.